modules/phase_field/include/materials/CrossTermBarrierFunctionMaterial.h

```diff
@@ -0,0 +1,49 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef CROSSTERMBARRIERFUNCTION_H
+#define CROSSTERMBARRIERFUNCTION_H
+
+#include "Material.h"
+#include "DerivativeMaterialInterface.h"
+
+// Forward Declarations
+class CrossTermBarrierFunctionMaterial;
+
+template<>
+InputParameters validParams<CrossTermBarrierFunctionMaterial>();
+
+/**
+ * CrossTermBarrierFunctionMaterial is a constraint kernel that acts on one of the eta_i variables to
+ * enforce \f$ \sum_n g_i(\eta_i) \equiv 1 \f$.
+ */
+class CrossTermBarrierFunctionMaterial : public DerivativeMaterialInterface<Material>
+{
+public:
+  CrossTermBarrierFunctionMaterial(const std::string & name, InputParameters parameters);
```

Comment:
Would you mind updating your code to use the new constructors, i.e.
```
CrossTermBarrierFunctionMaterial(const InputParameters & parameters);
```
?  It looks like it won't with marmot as-is...